src/config.c

```diff
@@ -607,60 +619,268 @@ int parseCommandLine(int argc,char **argv,ParodusCfg * cfg)
 }
 
 /*
-* call parodus create/acquisition script to create new auth token, if success then calls 
-* execute_token_script func with args as parodus read script.
+* @brief Initialize curl object with required options. create newToken using libcurl.
+* @param[out] newToken auth token obtained from JWT curl response
+* @param[in] len total token size
+* @param[in] r_count Number of curl retries on ipv4 and ipv6 mode during failure
 */
 
-void createNewAuthToken(char *newToken, size_t len)
+int createNewAuthToken(char *newToken, size_t len, int r_count)
 {
-	//Call create script
-	char output[12] = {'\0'};
-	execute_token_script(output,get_parodus_cfg()->token_acquisition_script,sizeof(output),get_parodus_cfg()->hw_mac,get_parodus_cfg()->hw_serial_number);
-  	if (strlen(output)>0  && strcmp(output,"SUCCESS")==0)
+	CURL *curl;
+	CURLcode res;
+	CURLcode time_res;
+	struct curl_slist *list = NULL;
+	struct curl_slist *headers_list = NULL;
+
+	char *mac_header = NULL;
+	char *serial_header = NULL;
+	char *uuid_header = NULL;
+	char *transaction_uuid = NULL;
+	double total;
+
+	struct url_data data;
+	data.size = 0;
+
+	curl = curl_easy_init();
+	if(curl)
 	{
-		//Call read script 
-		execute_token_script(newToken,get_parodus_cfg()->token_read_script,len,get_parodus_cfg()->hw_mac,get_parodus_cfg()->hw_serial_number);
-	}	
-	else 
+		data.data = (char *) malloc(sizeof(char) * MAX_TOKEN_LEN);
+		if(NULL == data.data)
+		{
+			ParodusError("Failed to allocate memory.\n");
+			return -1;
+		}
+		data.data[0] = '\0';
+
+		createCurlheader(mac_header, serial_header, uuid_header, transaction_uuid, list, &headers_list);
+
+		curl_easy_setopt(curl, CURLOPT_URL, TOKEN_SERVER_URL);
+		curl_easy_setopt(curl, CURLOPT_TIMEOUT, CURL_TIMEOUT_SEC);
+		curl_easy_setopt(curl, CURLOPT_INTERFACE, get_parodus_cfg()->webpa_interface_used);
```

Comment:
If config interface is empty "" length 0, then this SET call should not be made so that default NULL, use whatever the TCP stack finds suitable

Reply:
incorporated.